### PR TITLE
Replace connection exception with response errors

### PR DESCRIPTION
**Pull request by ShanniLi**
r @anson627

### CODE REVIEW

**Comment by anson627** [comment]
:+1:

File: tchannel-core/src/main/java/com/uber/tchannel/api/SubChannel.java [hunk 1]
```diff
@@ -225,4 +227,28 @@ protected ResponseRouter prepare(
     ) throws TChannelError {
         return send(request, null, 0);
     }
+
+    public <V> ListenableFuture<V> createError(Request request, ErrorType errorType, Throwable throwable) {
+        TFuture<V> future = TFuture.create();
+        ArgScheme argScheme = request.getArgScheme();
+        Response response = Response.build(argScheme, new ErrorResponse(
```
Comment (anson627) [inline tchannel-core/src/main/java/com/uber/tchannel/api/SubChannel.java]: maybe response builder should take care of building different Response based on different ArgScheme?

File: tchannel-core/src/main/java/com/uber/tchannel/messages/ErrorResponse.java [hunk 2]
```diff
@@ -30,12 +30,22 @@
...
     public ErrorResponse(long id, ErrorType errorType, String message) {
         this.id = id;
         this.errorType = errorType;
         this.message = message;
         this.type = FrameType.Error;
+        this.throwable = null;
+    }
+
+    public ErrorResponse(long id, ErrorType errorType, Throwable throwable) {
+        this.id = id;
```
Comment (anson627) [inline tchannel-core/src/main/java/com/uber/tchannel/messages/ErrorResponse.java]: you can have a private base constructor, and have both call it with different parameters.

File: tchannel-core/src/main/java/com/uber/tchannel/api/SubChannel.java (hunk 1 above)
Comment (ShanniLi) [inline tchannel-core/src/main/java/com/uber/tchannel/api/SubChannel.java]: Refactored part of the code to ResponseRouter.

**Pull request merged**